columnflow/config_util.py

```diff
@@ -235,3 +235,30 @@ def kwargs_fn(categories):
                     parent_cat.add_category(cat)
 
     return n_created_categories
+
+
+def verify_config_processes(config: od.Config, warn: bool = False) -> None:
+    """
+    Verifies for all datasets contained in a *config* object that the linked process is covered by
+    any process object registered in *config* and raises an exception if not. If *warn* is *True*,
+    a warning is printed instead.
+    """
+    missing_pairs = []
+    for dataset in config.datasets:
+        if not config.has_process(process := dataset.processes.get_first()):
+            missing_pairs.append((dataset, process))
+
+    # nothing to do when nothing is missing
+    if not missing_pairs:
+        return
+
+    # build the message
+    msg = f"found {len(missing_pairs)} dataset(s) whose process is not registered in the '{config.name}' config:"
+    for dataset, process in missing_pairs:
+        msg += f"\n  dataset '{dataset.name}' -> process '{process.name}'"
+
+    # warn or raise
+    if warn:
+        print(f"{law.util.colored('WARNING:', 'red')} {msg}")
+    else:
+        raise Exception(msg)
```

Comment:
Raising an `Exception` here is very generic. Maybe a `ValueError` might be more specific?

Reply:
Not sure if `ValueError` fits better. Maybe at some point, we should have a few (not too many) custom exceptions like `CFConfigError` or so :)
But I wouldn't tackle that in this PR.